src/scripts/mod.rs

```diff
@@ -0,0 +1,162 @@
+//! Scripting engine to run scripts based on tags and given ip-ports
+//! First read config file
+//!
+use anyhow::{anyhow, Result};
+use std::net::IpAddr;
+use std::path::PathBuf;
+use subprocess::{Exec, ExitStatus};
+use which;
+
+// struct ScriptEngine {
+//     nmap: Bin,
+//     python: Bin,
+// }
+
+// impl ScriptEngine {
+//     fn build(&self, interpreter: String, arguments: Vec<String>, ip: String, ports: Vec<u16>) {
+
+//     }
+//     fn execute(script: Script, addr: &str, ports: Vec<_>, arguments: Vec<String>) {
+
+//     }
+// }
+
+pub struct Script {
+    command: String,
+    path: Option<PathBuf>,
+    arguments: Vec<String>,
+    ip: IpAddr,
+    ports: Vec<u16>,
+    tags: Option<Vec<String>>,
+}
+
+pub enum Executable {
+    Nmap,
+    Python,
+    Bash,
+    Perl,
+}
+
+impl Script {
+    pub fn build(
+        command: Executable,
+        path: Option<PathBuf>,
+        arguments: Vec<String>,
+        ip: IpAddr,
+        ports: Vec<u16>,
+        tags: Option<Vec<String>>,
+    ) -> Self {
+        let mut command_match = String::from("nmap");
+        match command {
+            Nmap => command_match = "nmap".into(),
+            _ => {}
+        }
+
+        // Main command to use ex. nmap, python, perl, bash
+        Self {
+            // command like nmap, or python
+            command: command_match,
+            // command: command,
+
+            // optional path to the executable
+            path: path,
+
+            // user given arguments on commandline after --
+            arguments: arguments,
+            ip: ip,
+            ports: ports,
+
+            /*  script tags stored in file header? We can use toml to read it into a struct after parsing the headers manually first
```

Comment:
We should actually do something like this, ideally:

```
#! Shebang thingy here
# RSE START
# Tags: Core, HTTP, Web, HTTPS
# Ports: 80, 443
# RSE END
```

That way we don't have to parse "maybe line 1 is a shebang?" and "does their comments end after 5 lines or 4?"

Also, if we use TOML like structure within them we can easily parse for the struct.

And searching for `RSE START` and `RSE END` are easier.

Use "end" instead of "stop" because "stop" is physically closer to "start" (same first 2 letters, small minor thing but it might help people read the code)

Reply:
We could do that, or just use a struct with predefined `Option`al fields, and parse from toml to that struct.